File: arrays/sliding-window/3090.ts
function maximumLengthSubstring(s: string): number {
    let pL = 0;
    let pR = 0;
    let max = 1;
    let charCount = new Map<string, number>();
    charCount.set(s[0], 1);

    while (pR < s.length - 1) {
        pR++;

        if (charCount.get(s[pR])) {
            charCount.set(s[pR], charCount.get(s[pR])! + 1);
        } else {
            charCount.set(s[pR], 1);
        }

        while (charCount.get(s[pR])! == 3) {
            const charLeft = s[pL];

            charCount.set(charLeft, charCount.get(charLeft)! - 1);
            pL++;
        }

        max = Math.max(max, pR - pL + 1);
    }

    return max;
};

File: arrays/sliding-window/3090.cs
using System.Formats.Tar;
using System.Globalization;

public class Solution {
    public int MaximumLengthSubstring(string s) {
        int pl = 0;
        int pr = 0;
        int _max = 1;
        var map = new Dictionary<int, string>();

        map[s[0]] = 1;

        while (pr < (s.Length()-1))
        {
            pr++;
            if(map[s[pr]])
            {
                map[s[pr]] += 1;
            }
            else
            {
                map[s[pr]] = 1;
            }

            while (map[s[pr]] == 3)
            {
                map[s[pl]] = -1;
                pl++;
            }
            
            if (_max < pr - pl + 1)
            {
                _max = pr-pl+1;
            }
        }

        return _max;
    }
}